### ISSUE TITLE
No detection during test time

### ISSUE THREAD

**Issue by wendyzzzw** [avanetten/simrdwn#38]
I was trying to reproduce the result with COWC, however, there are no cars detected during test time while the loss curve during training time looks reasonable:

![yolt_loss_plot_tiny_cowc](https://user-images.githubusercontent.com/10052602/57990905-e8ef7600-7a5f-11e9-97ed-d12775f08ee1.png)

I was using yolo-v2-tiny config and weights, and I was using Tesla K80 on GCP with CUDA Version: 10.1. Any idea on what's going wrong??

**Comment by avanetten** [comment]
It's hard to determine what might be happening without further details.  What test data are you using, and what parameters are used for the test command?

**Comment by wendyzzzw** [comment]
@avanetten I am using the COWC data, here are my commands for training and testing:

`python simrdwn/core/simrdwn.py \
	--framework yolt2 \
	--mode train \
	--outname yolov2_cars \
	--yolt_cfg_file yolov2tiny.cfg  \
	--weight_dir simrdwn2/yolt2/input_weights \
	--weight_file yolov2-tiny.weights \
	--yolt_train_images_list_file cowc_yolt_train_list.txt \
	--label_map_path simrdwn2/train_data/class_labels_car.pbtxt \
	--nbands 3 \
	--max_batches 10000 \
	--batch_size 64 \
	--subdivisions 32 \
	--gpu 0`


`python simrdwn/core/simrdwn.py \
	--framework yolt2 \
	--mode test \
	--outname yolov2_cars \
	--label_map_path 'simrdwn/train_data/class_labels_car.pbtxt' \
	--train_model_path 'train_yolt2_yolov2_cars_2019_05_30_06-30-51' \
	--weight_file yolov2tiny_final.weights \
	--yolt_cfg_file yolov2tiny.cfg \
	--testims_dir 'test_images/cowc'  \
	--keep_test_slices 0 \
	--test_slice_sep __ \
	--test_make_legend_and_title 0 \
	--edge_buffer_test 1 \
	--test_box_rescale_frac 1 \
	--plot_thresh_str 0.01 \
	--slice_sizes_str 416 \
	--slice_overlap 0.2 \
	--alpha_scaling 1 \
	--show_labels 1 `

**Comment by wendyzzzw** [comment]
@avanetten also, sometimes there is a bug showing up during test time: pd.read_csv cannot find file "results/test_xxxxxx/cars.txt"

This bug can be fixed by just re-making yolt2. I am not sure why this is happening.

**Comment by MhmdDimassi** [comment]
I have the same problem, did you fix it?
@wendyzzzw

**Comment by avanetten** [comment]
Can you print out the output of the test script?  The issue in pd.read_csv() is usually caused because there was an error further upstream and the inference script did not run correctly

**Comment by MhmdDimassi** [comment]
Yes sure, i was print it now in my issue: https://github.com/avanetten/simrdwn/issues/44#issue-455175193

**Comment by ssunlin** [comment]
> I was trying to reproduce the result with COWC, however, there are no cars detected during test time while the loss curve during training time looks reasonable:
> 
> ![yolt_loss_plot_tiny_cowc](https://user-images.githubusercontent.com/10052602/57990905-e8ef7600-7a5f-11e9-97ed-d12775f08ee1.png)
> 
> I was using yolo-v2-tiny config and weights, and I was using Tesla K80 on GCP with CUDA Version: 10.1. Any idea on what's going wrong??

 @avanetten I have encountered a similar situation.
![yolt_loss_plot](https://user-images.githubusercontent.com/17522209/65389704-9adedc00-dd8b-11e9-84b7-c3091c727d17.png)
I think , even though the model don't convergence correctly, there should be some wrong boxes in the output file,while there is nothing. It makes me confused.

**Comment by Spiruel** [comment]
I have the same problem.  @MhmdDimassi @wendyzzzw did you manage to fix it?

**Comment by MhmdDimassi** [comment]
> I have the same problem. @MhmdDimassi @wendyzzzw did you manage to fix it?

Can you please put the test command that you use

**Comment by Spiruel** [comment]
> > I have the same problem. @MhmdDimassi @wendyzzzw did you manage to fix it?
> 
> Can you please put the test command that you use

`python simrdwn/core/simrdwn.py --framework yolt2 --mode test --outname dense_cowc --label_map_path class_labels_car.pbtxt --train_model_path /simrdwn/results/train_yolt2_dense_cars_2019_11_05_16-40-12 --weight_file ave_dense_final.weights --yolt_cfg_file ave_dense.cfg --testims_dir cowc/Utah_AGRC  --keep_test_slices 0 --test_slice_sep __ --test_make_legend_and_title 0 --edge_buffer_test 1 --test_box_rescale_frac 1 --plot_thresh_str 0.2 --slice_sizes_str 416 --slice_overlap 0.2 --alpha_scaling 1 --show_labels 1`

**Comment by avanetten** [comment]
This is likely an issue with YOLO and updates to architectures (see https://github.com/avanetten/simrdwn/issues/31#issuecomment-501925778).  You may need to experiment with the yolt2 [Makefile](https://github.com/avanetten/simrdwn/blob/master/yolt2/Makefile).

**Comment by Spiruel** [comment]
> This is likely an issue with YOLO and updates to architectures (see [#31 (comment)](https://github.com/avanetten/simrdwn/issues/31#issuecomment-501925778)). You may need to experiment with the yolt2 [Makefile](https://github.com/avanetten/simrdwn/blob/master/yolt2/Makefile).

OK thanks I will loo into this. Do you have any working COWC weights that I may use to help debug, so I can avoid training all over again?
